From c8decd4241e7513399b492cec548f1ceb5712843 Mon Sep 17 00:00:00 2001
From: Andrew Wesie <awesie@gmail.com>
Date: Mon, 23 Jul 2018 19:30:54 -0700
Subject: [PATCH 2/9] Unsupported stub for MemoryWorkingSetExInformation.

Implemented in Windows Vista. This will cause programs to fail if Wine version
is higher than Windows XP / 2003.
---
 dlls/ntdll/virtual.c | 1 +
 include/winternl.h   | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/dlls/ntdll/virtual.c b/dlls/ntdll/virtual.c
index 36f0aef..a39ca94 100644
--- a/dlls/ntdll/virtual.c
+++ b/dlls/ntdll/virtual.c
@@ -3135,6 +3135,7 @@ NTSTATUS WINAPI NtQueryVirtualMemory( HANDLE process, LPCVOID addr,
 
         UNIMPLEMENTED_INFO_CLASS(MemoryWorkingSetList);
         UNIMPLEMENTED_INFO_CLASS(MemoryBasicVlmInformation);
+        UNIMPLEMENTED_INFO_CLASS(MemoryWorkingSetExInformation);
 
         default:
             FIXME("(%p,%p,info_class=%d,%p,%ld,%p) Unknown information class\n",
diff --git a/include/winternl.h b/include/winternl.h
index 2ea22ed..8924d76 100644
--- a/include/winternl.h
+++ b/include/winternl.h
@@ -1038,7 +1038,8 @@ typedef enum _MEMORY_INFORMATION_CLASS {
     MemoryBasicInformation,
     MemoryWorkingSetList,
     MemorySectionName,
-    MemoryBasicVlmInformation
+    MemoryBasicVlmInformation,
+    MemoryWorkingSetExInformation
 } MEMORY_INFORMATION_CLASS;
 
 typedef struct _MEMORY_SECTION_NAME
-- 
2.7.4

